lib/errors/index.ts

```diff
@@ -0,0 +1,94 @@
+import * as http from 'http';
```

Comment:
If we are only importing for the type, doesn't type script have some sort of type import? Feels a bit heavy to import all of http just for the few types.

Reply:
TypeScript usually remove the imports if we are only using types, but there's no guarantee. I'm used to import whatever and let the tree shaking do the work, but you're totally right, import type does not add any overhead at runtime (because types are stripped at compile-time).